src/base/static/js/models/attachment-model.js

```diff
@@ -53,7 +53,9 @@ module.exports = Backbone.Model.extend({
       progressHandler = Util.wrapHandler("progress", this, options.progress),
       myXhr = $.ajaxSettings.xhr();
 
-    formData.append("file", blob);
+    if (blob) {
+      formData.append("file", blob);
+    }
```

Comment:
Sometimes the application will be in a state where the attachment model's `blob` attribute is undefined. As a result, when we "delete" an attachment model, a PATCH request is sent with a `file` mapped to `undefined` in the request's body.

I'm still not sure when the attachment model has an `undefined` value for `blob`, but when it does happen, it causes the new 1.8 api to return a 400. This is because the 1.8 api performs an extra validation step, where it will reject the request if the `file` in the request's body has an invalid value, which in this case is the string `"undefined"`.